### PR TITLE
IFCTool: Finalizing Flood Filling

### PR DESCRIPTION
**Pull request by mkeshavarzi**
This is a draft PR for finalizing tests and resolving bugs for the 3D floodfilling alforithm. Tests would include unit tests and tests with simple three.js bounding boxes and imported obj bounding boxes.

### CODE REVIEW

**Comment by pablo-mayrgundter** [comment]
For draft PRs, it's actually a status set on the PR, not in the description.  Toggle it here:

image

That will make it show up differently in the queue

File: .gitignore [hunk 1]
```diff
@@ -102,3 +102,4 @@ dist
 
 # TernJS port file
 .tern-port
+lib/src/.DS_Store
```
Comment (pablo-mayrgundter) [inline .gitignore:105]: Huh, not sure why we have all these in the .gitignore.. i probably chose the wrong template.  can you just replace the whole thing with this and we'll see how it goes..
```
node_modules/
.DS_Store
logs
*.log
```

File: lib/src/extractLevels.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
+import fs from 'fs'
```
Comment (pablo-mayrgundter) [inline lib/src/extractLevels.js:1]: remove from this PR

File: lib/src/extractLevels.test.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import fs from 'fs'
```
Comment (pablo-mayrgundter) [inline lib/src/extractLevels.test.js:1]: remove from this PR

File: lib/src/testdata/testSimpleBB.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable */
```
Comment (pablo-mayrgundter) [inline lib/src/testdata/testSimpleBB.js:1]: move this up a dir.  this dir should not contain executable code

File: package.json [hunk 5]
```diff
@@ -59,5 +59,8 @@
   },
   "description": "Command line tool for working with IFC models.  It extracts IFC elements by ID or types, and exports as JSON or CSV (via [json2csv](https://www.npmjs.com/package/json2csv)).",
   "author": "info@bldrs.ai",
-  "license": "MIT"
+  "license": "MIT",
+  "dependencies": {
```
Comment (pablo-mayrgundter) [inline package.json:63]: remove this to lib/package.json

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]

**Comment by pablo-mayrgundter** [comment]
Heya, how's this going?

**Comment by pablo-mayrgundter** [comment]
Please merge from upstream then lmk

**Comment by pablo-mayrgundter** [comment]
Hey Mohammed, please address open comments from my review

**Comment by mkeshavarzi** [comment]
> Hey Mohammed, please address open comments from my review

Hey Pablo - comments are addressed but "three" is still failing to import in tests.js.

File: .gitignore (hunk 1 above)
Comment (mkeshavarzi) [reply .gitignore]: Cleaned .gitignore

File: lib/src/extractLevels.test.js (hunk 3 above)
Comment (mkeshavarzi) [reply lib/src/extractLevels.test.js]: Removed

File: lib/src/testdata/testSimpleBB.js (hunk 4 above)
Comment (mkeshavarzi) [reply lib/src/testdata/testSimpleBB.js]: Moved up a dir.

File: package.json (hunk 5 above)
Comment (mkeshavarzi) [reply package.json]: Removed.

**Pull request closed without merging**